src/sdk/autodetectcompilers.cpp

```diff
@@ -40,109 +45,240 @@ AutoDetectCompilers::AutoDetectCompilers(wxWindow* parent)
+                                cItem.bDetected = false;
+                                cItem.eHighlight = CompilerHighlightColor::eHighlightRed;
+                            }
 
-                    // Assume the user did the setup on purpose, so reset the old settings anyways:
-                    compiler->SetMasterPath(path);
+                            // Assume the user did the setup on purpose, so reset the old settings anyways:
+                            compiler->SetMasterPath(path);
+                        }
+                    }
                 }
-            }
-            else // no built-in, but user-defined (i.e. copied) compiler
-            {
-                // Check, if the master path is valid:
-                if ( !path.IsEmpty() && wxFileName::DirExists(path_no_macros) )
+                else // no built-in, but user-defined (i.e. copied) compiler
                 {
-                    list->SetItem(idx, 1, _("User-defined")); // OK
-                    highlight = 0;
-                }
+                    // Check, if the master path is valid:
+                    if ( !path.IsEmpty() && wxFileName::DirExists(path_no_macros) )
+                    {
+                        cItem.wxsStatus = _("User-defined");
+                        cItem.bDetected = true;
+                        cItem.eHighlight = CompilerHighlightColor::eHighlightGreen;
+                    }
+                    else
+                    {
+                        cItem.wxsStatus = _("User-defined masterpath issue");
+                        cItem.bDetected = false;
+                        cItem.eHighlight = CompilerHighlightColor::eHighlightRed;
+                    }
+                }
+            }
+            else
+            {
+                cItem.wxsStatus = _("No compiler");
+                cItem.wxsCompilerPath = _("N/A");
+                cItem.bDetected = true;
+                cItem.eHighlight = CompilerHighlightColor::eHighlightGreen;
             }
 
-            if (highlight == 1)
-                list->SetItemBackgroundColour(idx, *wxRED);
-            else if (highlight == -1)
-                list->SetItemTextColour(idx, *wxLIGHT_GREY);
+            if (defaultCompilerID.IsSameAs(currentCompilerID))
+            {
+                wxStaticText* controlLblDefCompiler = XRCCTRL(*this, "lblDefCompiler", wxStaticText);
+                if (cItem.bDetected)
+                {
+                    wxColour  colorDefault =  XRCCTRL(*this, "ID_STATICTEXT1", wxStaticText)->GetBackgroundColour();
+
+                    controlLblDefCompiler->SetLabel(cItem.wxsCompilerName);
+                    controlLblDefCompiler->SetBackgroundColour(colorDefault);
+                }
+                else
+                {
+                    controlLblDefCompiler->SetLabel(wxString::Format(_("INVALID: %s"),cItem.wxsCompilerName));
+                    controlLblDefCompiler->SetBackgroundColour(*wxRED);

```

Comment:
This is unacceptable. Changing just the background breaks on non-default themes. Please remove this colour changing.

Reply:
The existing code does this as well.

This is the existing code for lines 120 to 123.
            if (highlight == 1)
                list->SetItemBackgroundColour(idx, *wxRED);
            else if (highlight == -1)
                list->SetItemTextColour(idx, *wxLIGHT_GREY);

Reply:
So the existing code is also wrong. It is not a valid excuse.